manifests/init.pp

```diff
@@ -171,16 +192,18 @@
 
   Class["keycloak::datasource::${datasource_driver}"]~>Class['keycloak::service']
 
-  keycloak_conn_validator { 'keycloak':
-    keycloak_server => 'localhost',
-    keycloak_port   => $http_port,
-    use_ssl         => false,
-    timeout         => 60,
-    test_url        => '/auth/realms/master/.well-known/openid-configuration',
-    require         => Class['keycloak::service'],
+# Keycloak conn validator is active only for HTTP connection, as HTTPS connection works only with Puppet certs and not custom keystore (Ticket SERVER-1543)
+  if ! $keycloak::https {
```

Comment:
Why not set `use_ssl => $keycloak::https` ?   Could also update port to point at either `$http_port` or `$https_port` , assuming SSL enabled support works.  Having `keycloak_conn_validator` is important for the custom types/providers.

Reply:
I tried it, but as I put in description, Keycloak_conn_validator will always fails, as it tries to connect using Puppet node SSL certificate, and not custom SSL truststore used in Keycloak. 
At the moment according to the ticket mentioned this is not supported (it's an issue of Puppet::Network::HttpPool.http_instance )

Reply:
Ah sorry missed that in description.  Using Puppet's HTTP stack was done because something copied from PuppetDB module, maybe at some point need to revisit and use standard `net/http` or some other HTTP communication stack.